pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -450,16 +450,24 @@ func (p *provider) Create(machine *v1alpha1.Machine, update cloud.MachineUpdater
 	return &azureVM{vm: &vm, ipAddresses: ipAddresses, status: status}, nil
 }
 
-func (p *provider) Delete(machine *v1alpha1.Machine, update cloud.MachineUpdater, instance instance.Instance) error {
+func (p *provider) Delete(machine *v1alpha1.Machine, update cloud.MachineUpdater) error {
 	config, _, err := p.getConfig(machine.Spec.ProviderConfig)
 	if err != nil {
 		return fmt.Errorf("failed to parse MachineSpec: %v", err)
 	}
 
-	glog.Infof("deleting VM %q", machine.Name)
-	if err = deleteVMsByMachineUID(context.TODO(), config, machine.UID); err != nil {
-		return fmt.Errorf("Is failed to remove public IP addresses of machine %q: %v", machine.Name, err)
+	_, err = p.Get(machine)
+	if err != nil {
+		if err != cloudprovidererrors.ErrInstanceNotFound {
```

Comment:
Yes. Looking at the flow it seems what happens is:

* Initial Get succeeds, because otherwise we never execute the `Create`
* We execute the `Create` and fail
* Every subsequent `Get` fails because the config of the vm is invalid

I guess we must call the `DeleteVM` always except if we get a `ErrInstanceNotFound`